build-coordinator/src/main/java/org/jboss/pnc/core/builder/datastore/DatastoreAdapter.java

```diff
@@ -106,11 +106,41 @@ public void storeResult(BuildTask buildTask, BuildResult buildResult) throws Dat
             log.debugf("Storing results of buildTask [%s] to datastore.", buildTask.getId());
             datastore.storeCompletedBuild(buildRecordBuilder, buildTask.getBuildRecordSetIds());
         } catch (Exception e) {
-            storeResult(buildTask, e);
+            storeErroneousResult(buildTask, e);
+        }
+    }
+
+
+    public void storeFailedResult(BuildTask buildTask, BuildResult buildResult) throws DatastoreException {
```

Comment:
There's indeed some code duplucation. I'll reduce it. For now I think that two methods are clearer, but I'll try out merging too. (after some tests start to show that it works ;) )

Thank you for reviewing the changes!